src/main/java/reposense/parser/CsvParser.java

```diff
@@ -0,0 +1,135 @@
+
+        try {
+            // Skips first line, which is the header row
+            final Collection<String> lines = Files.lines(path).skip(SKIP_FIRST_LINE).collect(Collectors.toList());
+
+            for (final String line : lines) {
+                processLine(repoConfigurations, sinceDate, untilDate, line, lineNumber);
+                lineNumber++;
+            }
+
+        } catch (IOException ioe) {
+            throw new IOException(MESSAGE_UNABLE_TO_READ_CSV_FILE, ioe);
+        }
+
+        return repoConfigurations;
+    }
+
+    /**
+     * Extracts {@code Author} information from the {@code line}.
+     */
+    private static void processLine(List<RepoConfiguration> repoConfigurations,
+            Date sinceDate, Date untilDate, String line, int lineNumber) {
+        if (line.isEmpty()) {
+            return;
+        }
+
+        String[] elements = line.split(Constants.CSV_SPLITTER);
+
+        if (elements.length < GITHUB_ID_POSITION) {
+            // Warns malformed line and skips it
+            logger.warning(String.format(MESSAGE_MALFORMED_LINE_FORMAT, lineNumber, line));
+            return;
+        }
+
+        String organization = elements[ORGANIZATION_POSITION];
+        String repositoryName = elements[REPOSITORY_NAME_POSITION];
+        String branch = elements[BRANCH_POSITION];
+
+        RepoConfiguration config = new RepoConfiguration(organization, repositoryName, branch);
+        int index = repoConfigurations.indexOf(config);
+
+        if (index != -1) {
+            config = repoConfigurations.get(index);
+        } else {
+            repoConfigurations.add(config);
+            config.setSinceDate(sinceDate);
+            config.setUntilDate(untilDate);
+        }
+
+        Author author = new Author(elements[GITHUB_ID_POSITION]);
+        config.getAuthorList().add(author);
+        setDisplayName(elements, config, author);
+        setAlias(elements, config, author);
+    }
+
+    private static void setDisplayName(String[] elements, RepoConfiguration config, Author author) {
+        boolean isDisplayNameInElements = elements.length > DISPLAY_NAME_POSITION
+                && !elements[DISPLAY_NAME_POSITION].isEmpty();
+
+        if (isDisplayNameInElements) {
+            //Not empty, take the supplied value as Display Name
+            config.getAuthorDisplayNameMap().put(author, elements[DISPLAY_NAME_POSITION]);
+        } else {
+            //Use GitHub Id as Display Name
+            config.getAuthorDisplayNameMap().put(author, author.getGitId());
+        }
+    }
+
+    private static void setAlias(String[] elements, RepoConfiguration config, Author author) {
+        //Always use GitHub Id as an alias
+        config.getAuthorAliasMap().put(elements[GITHUB_ID_POSITION], author);
+        // If more aliases are provided, use them as well
+        boolean isAliasPositionInElements = elements.length > ALIAS_POSITION
+                && !elements[ALIAS_POSITION].isEmpty();
```

Comment:
When you add comments like this, the reader will take the comment as applicable to the code block immediately below it. That's not the case here. You are better of adding a header comment to the whole method instead of these inline comments.